org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/launch/BaseLaunchRequestJob.java

```diff
@@ -46,24 +47,37 @@ public Void runInToolingApi(CancellationTokenSource tokenSource, IProgressMonito
     }
 
     protected final void executeLaunch(CancellationTokenSource tokenSource, final IProgressMonitor monitor) throws Exception {
-        // todo (etst) close streams when done
-
-        BuildExecutionParticipants.activateParticipantPlugins();
+        BuildExecutionParticipants buildParticipants = BuildExecutionParticipants.create(CorePlugin.extensionManager().loadBuildExecutionParticipants());
+        
         monitor.beginTask(getJobTaskName(), IProgressMonitor.UNKNOWN);
 
         ProcessDescription processDescription = createProcessDescription();
         RunConfiguration runConfig = getRunConfig();
+        
         InternalGradleBuild gradleBuild = CorePlugin.internalGradleWorkspace().getGradleBuild(runConfig.getProjectConfiguration().getBuildConfiguration());
+
+        buildParticipants.preAction(runConfig.getProjectConfiguration().getProjectDir(), runConfig.getTasks(), monitor);
```

Comment:
Should we pass more into the preBuild method?  To let participants have more access?

Reply:
which parameters you want to pass to preBuild method? Most class were moved into internal package, like processDescription and RunConfiguration, That means we will add more basic type parameters.